### ISSUE TITLE
refactor how search-index works

### ISSUE THREAD

**Issue by peterbe** [mdn/yari#3139]
At the moment, the `search-index.json` is generated when you start the server (Express server). It's generated in a `Worker` to avoid blocking the main thread. 
Because it's directly imported by the server, whenever `nodemon` restarts, it triggers the `Worker` to rebuild the search index again. This is often quite annoying because it takes 2-4 seconds which is a waste of CPU when it probably hasn't changed. 
It also only works for `en-US` (i.e. `CONTENT_ROOT`).

It's nice that the file is generated quite quickly. A writer might type `yarn start` and quickly move to the browser (localhost:5000) and quickly reach for the search widget which triggers the XHR request. 

But it might be better to JIT it. Every time. That means it should be able to pick up any new documents to edited titles as soon as possible. 

Another thing we might want to consider is to write a custom piece of code that does the traversing. Because all it needs is the `title` and `url` from the is a fraction of what `Document.read()` does. Perhaps it's premature optimization but we could make it read a stream of each file till it has encountered the *second* front-matter marker (`---`) and bail. 

Basically, `yarn build` (with or without `CONTENT_TRANSLATED_ROOT`) should always create those `client/build/**/search-index.json` files.
And when you're running a local dev/preview server it should be possible to XHR request `/$locale/search-index.json` and it would be nice if this was reasonably speedy.
